src/common/regions.ts

```diff
@@ -0,0 +1,17 @@
+// All regions for WoW
+export const ALL_REGIONS = ['EU', 'US', 'TW', 'KR', 'CN'] as const;
+export type Region = typeof ALL_REGIONS[number];
+
+// WoWAnalyzer/Blizzard API supported regions
+export const SUPPORTED_REGIONS = ['EU', 'US', 'TW', 'KR'] as const;
+export type SupportedRegion = typeof SUPPORTED_REGIONS[number];
+
+export function isRegion(region: string | undefined): region is Region {
+  // the 'as' below is just to allow the use of .includes
+  return ALL_REGIONS.includes(region as Region);
+}
+
+export function isSupportedRegion(region: string | undefined): region is SupportedRegion {
```

Comment:
this should get a docstring and/or name update clarifying that this is related to API support, not analysis support.

```suggestion
/**
 * Not every region is supported by the Blizzard API (and, by extension, the bits of the WoWA API that use the Blizzard API). This is used to check that that a region is supported before sending a request.
 */
export function isSupportedApiRegion(region: string | undefined): region is SupportedRegion {
```